Source/WebCore/page/text-extraction/TextExtractionTypes.h

```diff
@@ -25,47 +25,51 @@
 
 #pragma once
 
+#include "CharacterRange.h"
 #include "FloatRect.h"
 #include "FloatSize.h"
 #include <wtf/Forward.h>
+#include <wtf/URL.h>
 
 namespace WebCore {
 namespace TextExtraction {
 
+struct Editable {
+    String label;
+    String placeholder;
+    bool isSecure { false };
+    bool isFocused { false };
+};
+
 struct TextItemData {
+    Vector<std::pair<URL, CharacterRange>> links;
```

Comment:
Does `std::pair` need the full class definitions? / Would forward decls not work?

Reply:
Hm.... I think I do need the implementations here, so that the size of `std::pair` is known. I think I can get away with removing `URL.h`, although that's just because it's already getting it from other unified sources, so I think it's probably best to keep including that explicitly.

Reply:
Yes I would say keep it in to avoid a future unified sources tragedy